inc/types/class-base.php

```diff
@@ -66,34 +66,45 @@ public function handle_authorisation() {
 			exit;
 		}
 
-		if ( empty( $_POST['_wpnonce'] ) ) {
-			return $this->render_form( $client );
-		}
+		/**
+		 * Filter the requirement for a user to authorize the oauth client application.
+		 *
+		 * @param bool Require user authorization
+		 * @param Client $client Client being authorized
+		 * @param string $scope Requested scope
+		 */
+		if ( apply_filters( 'oauth2.authorization.client_requires_user_authorization', true, $client, $scope ) ) {
```

Comment:
I'd prefer the filter be broken out to a separate line here, as it's quite long.